src/main/java/org/testng/internal/MethodInvocationHelper.java

```diff
@@ -288,7 +288,6 @@ private static void invokeWithTimeoutWithNewExecutor(ITestNGMethod tm, Object in
       ThreadTimeoutException exception = new ThreadTimeoutException("Method "
           + tm.getQualifiedName() + "()"
           + " didn't finish within the time-out " + realTimeOut);
-      exception.setStackTrace(exec.getStackTraces()[0]);

```

Comment:
I guess its perhaps because `org.testng.internal.MethodInvocationHelper#invokeWithTimeoutWithNewExecutor` is designed to just work with 1 thread due to this line

```java
IExecutor exec = ThreadUtil.createExecutor(1, tm.getMethodName());
```
So I guess there would never be more than 1 thread.

Moreover, if you look at `org.testng.internal.MethodInvocationHelper#invokeWithTimeoutWithNewExecutor` implementation, you would notice that it first interrupts all the running threads by calling `exec.stopNow();` and then trying to access the threads that the underlying factory would have. That explains why its always an empty array.

Lastly I would still like to understand what is the value add that we would be adding by copy pasting the stacktrace. The thread has been interrupted (and that too by TestNG). The error msg IMHO is what adds a bit more value.